gremlin-server/src/test/java/org/apache/tinkerpop/gremlin/server/GremlinServerIntegrateTest.java

```diff
@@ -293,6 +312,21 @@ private static SslContext createServerSslContext() {
         return scriptEngineConf;
     }
 
+    @Test
+    public void shouldPingChannelIfClientDies() throws Exception {
```

Comment:
You should be able test for a dead client by opening a raw WebSocket, or even TCP, connection and simply not responding to anything.

Or separately, just disable the idleWriteLimit (keepalive) and connect. Then it should timeout the client.

Reply:
good idea to just disable keepalive - that made it easier to write a test there.